Source/WebCore/dom/Element.cpp

```diff
@@ -2869,7 +2869,7 @@ void Element::updateEffectiveTextDirectionIfNeeded()
     if (!(parent && parent->usesEffectiveTextDirection()))
         return;
 
-    if (hasAttributeWithoutSynchronization(HTMLNames::dirAttr) || shadowRoot() || firstChild()) {
+    if (hasAttributeWithoutSynchronization(HTMLNames::dirAttr) || shadowRoot()) {
```

Comment:
Is it guaranteed that when an Element is `insertedIntoAncestor()`, all its descendants will also be re-`insertedIntoAncestor()`?

I tried different scenarios like moving a sub tree from a parent to another parent and removing a sub-tree from the Document and then inserting it back and they seem to work fine.

I could not recall why I added this condition in the first place.

Reply:
It is, the caller iterates all nodes in the subtree.